src/main/java/de/hpi/bpt/argos/storage/hierarchy/HierarchyBuilderImpl.java

```diff
@@ -0,0 +1,91 @@
+package de.hpi.bpt.argos.storage.hierarchy;
+
+import de.hpi.bpt.argos.core.Argos;
+import de.hpi.bpt.argos.storage.PersistenceAdapterImpl;
+import de.hpi.bpt.argos.storage.dataModel.entity.Entity;
+import de.hpi.bpt.argos.storage.dataModel.entity.VirtualRoot;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class HierarchyBuilderImpl implements HierarchyBuilder {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Make this class "final" or add a public constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2974)